tests/gtest/avifmetadatatest.cc

```diff
@@ -413,6 +414,32 @@ TEST(MetadataTest, MultipleExtendedXMPAndAlternativeGUIDTag) {
 
 //------------------------------------------------------------------------------
 
+// Regression test for https://github.com/AOMediaCodec/libavif/issues/1333.
+// Coverage for avifFixImageXMP().
+TEST(MetadataTest, XMPWithTrailingNullCharacter) {
+  testutil::AvifImagePtr jpg =
+      testutil::ReadImage(data_path, "paris_xmp_trailing_null.jpg");
+  ASSERT_NE(jpg, nullptr);
+  ASSERT_NE(jpg->xmp.size, 0u);
+  // avifJPEGRead() should strip the trailing null character.
+  ASSERT_EQ(std::memchr(jpg->xmp.data, '\0', jpg->xmp.size), nullptr);
+
+  // Append a zero byte to see what happens when encoded with libpng.
+  avifRWDataRealloc(&jpg->xmp, jpg->xmp.size + 1);
+  jpg->xmp.data[jpg->xmp.size] = '\0';
+  testutil::WriteImage(jpg.get(),
+                       (testing::TempDir() + "xmp_trailing_null.png").c_str());
```

Comment:
Optional: It would be better to construct the file with the trailing null byte and add it as a test data file. Then we don't need to worry about libpng export taking care of this as described by the comment at line 436.

Reply:
I believe libpng import may also strip the trailing null byte, so constructing a file to skip the export step will not bypass libpng taking of this entirely anyway.